src/index.d.ts

```diff
@@ -755,20 +768,103 @@ export function isPushPermissionGranted(callback: CallbackString): void;
+  export function syncCustomTemplates(): void;
 
   /**
-   * Fetches In Apps from server.
+   * Uploads Custom in-app templates and app functions to the server.
+   * 
+   * @param isProduction Provide `true` if templates must be sync in Productuon build/configuration.
+   */
+  export function syncCustomTemplatesInProd(isProduction: boolean): void;
+
+  /**
+   * Notify the SDK that an active custom template is dismissed. The active custom template is considered to be
+   * visible to the user until this method is called. Since the SDK can show only one InApp message at a time, all
+   * other messages will be queued until the current one is dismissed.
+   * 
+   * @param templateName The name of the active template
+   */
+  export function customTemplateSetDismissed(templateName: string): Promise<void>;
+
+  /**
+   * Notify the SDK that an active custom template is presented to the user
+   * 
+   * @param templateName The name of the active template
+   */
+  export function customTemplateSetPresented(templateName: string): Promise<void>;
+
+  /**
+   * Trigger a custom template action argument by name.
+   * 
+   * @param templateName The name of an active template for which the action is defined
+   * @param argName The action argument name
+   */
+  export function customTemplateRunAction(templateName: string, argName: string): Promise<void>;
+
+  /**
+   * Retrieve a string argument by name.
    *
-   * @param {function(err, res)} callback a callback with a boolean flag whether the fetching was successful
+   * @param templateName The name of an active template for which the argument is defined
+   * @param argName The action argument name
+   * 
+   * @returns The argument value or null if no such argument is defined for the template.
    */
-  export function fetchInApps(callback: Callback): void;
+  export function customTemplateGetStringArg(templateName: string, argName: string): Promise<string>;
 
   /**
-   * Deletes all images and gifs which are preloaded for inapps in cs mode
+   * Retrieve a number argument by name.
    *
-   * @param {boolean} expiredOnly to clear only assets which will not be needed further for inapps
+   * @param templateName The name of an active template for which the argument is defined
+   * @param argName The action argument name
+   * 
+   * @returns The argument value or null if no such argument is defined for the template.
    */
-  export function clearInAppResources(expiredOnly: boolean): void;
+  export function customTemplateGetNumberArg(templateName: string, argName: string): Promise<number>;
+
+  /**
+   * Retrieve a boolean argument by name.
+   *
+   * @param templateName The name of an active template for which the argument is defined
+   * @param argName The action argument name
+   * 
+   * @returns The argument value or null if no such argument is defined for the template.
+   */
+  export function customTemplateGetBooleanArg(templateName: string, argName: string): Promise<boolean>;
```

Comment:
Is it necessary to have getters for all different argument types? @vasct

Reply:
Yes, because they correlate to the same methods in the native sdk. The native methods verify that the argument is of the correct type before returning it and so the JS methods must also be called for the correct type. Also I think that it is good to know the type of the argument you are getting.